### PR TITLE
feat: Upgrade SDK v0.47 to v0.50

### PR DESCRIPTION
**Pull request by deepan95dev**
# Related Github tickets

- Upgrading SDK v0.47.4 to v0.50.2

# Background

_Upgraded cosmos sdk version of pigeon App by using the upgraded [paloma](https://github.com/palomachain/paloma/commit/bcdf4d248c51caa2898f58d4b42e0d91add59395) version_

# Testing completed

- [x] test coverage exists and has been updated
- [ ] tested in a private testnet

### CODE REVIEW

File: util/ion/broadcast_test.go [hunk 1]
```diff
@@ -29,6 +30,10 @@ type myFakeTx struct {
 	msgs []myFakeMsg
 }
 
+func (t myFakeTx) GetMsgsV2() ([]protov2.Message, error) {
+	return nil, nil // this is a hack for tests
+}
```
Comment (byte-bandit) [inline util/ion/broadcast_test.go:35]: Why exactly is this needed? Is this a new requirement on an interface from cosmos? If so, can you leave a link and some extended docs in the code ?

**Review by byte-bandit — CHANGES_REQUESTED** [review]
LGTM, I'm happy to see that so little change seems to be needed here. Blocking this now until the E2E tests are done. 

I will follow up with more info on that over here: https://github.com/palomachain/paloma/pull/1069

File: util/ion/broadcast_test.go (hunk 1 above)
Comment (deepan95dev) [reply util/ion/broadcast_test.go:35]: Just for the test case we needed this function, [refer](https://github.com/cosmos/cosmos-sdk/blob/9814f684b9dd7e384064ca86876688c05e685e54/types/mempool/mempool_test.go#L79) for similar case

**Comment by vishal-kanna** [comment]
@byte-bandit Please merge this PR no more changes from our side.

**Pull request closed without merging**